pkg/archive/archive.go

```diff
@@ -530,7 +530,21 @@ func createTarFile(path, extractDir string, hdr *tar.Header, reader io.Reader, L
 		}
 
 	case tar.TypeLink:
-		targetPath := filepath.Join(extractDir, hdr.Linkname)
+		//ln a b in container
+		//docker cp test:/tmp tmp_
+		//hdr.Linkname = "tmp/a"
+		//hdr.Name = "tmp_/b"
+		//--> lpath = tmp_/a -->targetPath =extractDir + lpath --> ln tmp_/a tmp_/b
+		lpath := hdr.Linkname
+		flinkName := strings.Split(hdr.Linkname, "/")
+		if len(flinkName) > 1 {
+			fName := strings.Split(hdr.Name, "/")
+			if len(fName) > 1 {
+				flinkName[0] = fName[0]
```

Comment:
I'm afraid filepath.Rel don't work for this.
```
        base := "tmp"
	p:="tmp_/a"
	rel, err := filepath.Rel(base, p)
	fmt.Printf("%q: %q %v\n", p, rel, err)
```
output is "../tmp_/a" , not "tmp_/a"